test/cppsuite/tests/cursor_microbenchmarks.cpp

```diff
@@ -0,0 +1,153 @@
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include "src/common/constants.h"
+#include "src/common/logger.h"
+#include "src/main/test.h"
+#include "src/util/execution_timer.h"
+
+namespace test_harness {
+/*
+ * This test aims to measure the number of instructions cursor API calls take. The test has measures
+ * in place to prevent background threads from taking resources:
+ *  - We set the sweep server interval to be greater than the test duration. This means it never
+ *    triggers.
+ *  - Logging, and the log manager thread are disabled per the connection open configuration.
+ *  - Prefetch, off by default
+ *  - Background compact, disabled by in_memory.
+ *  - Capacity server, disabled by in_memory.
+ *  - Checkpoint server, disabled by in_memory.
+ *  - Eviction, still runs but ideally we don't cross any threshold to trigger it.
+ *  - Checkpoint cleanup, disabled by in_memory.
+ *
+ * Additionally to avoid I/O the connection is set to in_memory.
+ */
+class cursor_microbenchmarks : public test {
+public:
+    cursor_microbenchmarks(const test_args &args) : test(args)
+    {
+        init_operation_tracker();
+    }
+
+    void
+    custom_operation(thread_worker *tc) override final
+    {
+        /* The test expects no more than one collection. */
+        testutil_assert(tc->collection_count == 1);
+
+        /* Assert that we are running in memory. */
+        testutil_assert(_config->get_bool(IN_MEMORY));
+
+        /* Test a single cursor insertion. */
+        execution_timer cursor_insert_timer("cursor_insert", test::_args.test_name, false);
+        execution_timer cursor_update_timer("cursor_update", test::_args.test_name, false);
+        execution_timer cursor_modify_timer("cursor_modify", test::_args.test_name, false);
+        execution_timer cursor_remove_timer("cursor_remove", test::_args.test_name, false);
+        execution_timer cursor_reset_timer("cursor_reset", test::_args.test_name);
+        execution_timer cursor_search_timer("cursor_search", test::_args.test_name);
+
+        collection &coll = tc->db.get_collection(0);
+        scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name);
+        auto key_count = coll.get_key_count();
+        tc->session->begin_transaction(tc->session.get(), NULL);
+
+        /* Benchmark cursor->search. Loop 1000 times to reduce noise. */
```

Comment:
Why is it the only op that uses a loop to reduce noise?

Reply:
Explained here: https://github.com/wiredtiger/wiredtiger/pull/10602#issuecomment-2137871687

Reply:
Thank you! Any attempt at increasing the cache size?

Reply:
We should be using the default which would be half of the systems RAM.

Reply:
Probably worth mentioning in the config file. We could potentially dynamically derive the cache size in case the underlying machine changes.